Biomass_speciesData.R

```diff
@@ -49,32 +52,45 @@ defineModule(sim, list(
+                    desc = "This describes the simulation time at which the first plot event should occur"),
     defineParameter(".plotInterval", "numeric", NA, NA, NA,
-                    "This describes the simulation time interval between plot events"),
+                    desc = "This describes the simulation time interval between plot events"),
+    defineParameter(".plots", "character", c("screen"), NA, NA,
+                    desc = paste("Passed to `types` in `Plots` (see `?Plots`).",
+                          "There are a few plots that are made within this module, if set.",
+                          "Note that plots (or their data) saving will ONLY occur at `end(sim)`.",
+                          "If `NA`, plotting is turned off completely (this includes plot saving).")),
     defineParameter(".saveInitialTime", "numeric", NA, NA, NA,
-                    "This describes the simulation time at which the first save event should occur"),
+                    desc = "This describes the simulation time at which the first save event should occur"),
     defineParameter(".saveInterval", "numeric", NA, NA, NA,
                     "This describes the simulation time interval between save events"),
     defineParameter(".sslVerify", "integer", as.integer(unname(curl::curl_options("^ssl_verifypeer$"))), NA_integer_, NA_integer_,
-                    paste("Passed to `httr::config(ssl_verifypeer = P(sim)$.sslVerify)` when downloading KNN",
+                    desc = paste("Passed to `httr::config(ssl_verifypeer = P(sim)$.sslVerify)` when downloading KNN",
                           "(NFI) datasets. Set to 0L if necessary to bypass checking the SSL certificate (this",
                           "may be necessary when NFI's website SSL certificate is not correctly configured).")),
     defineParameter(".studyAreaName", "character", NA, NA, NA,
                     "Human-readable name for the study area used. If NA, a hash of `studyAreaLarge` will be used."),
     defineParameter(".useCache", "character", "init", NA, NA,
                     desc = "Controls cache; caches the init event by default"),
     defineParameter(".useParallel", "numeric", parallel::detectCores(), NA, NA,
-                    "Used in reading csv file with fread. Will be passed to `data.table::setDTthreads`.")
+                    desc = "Used in reading csv file with fread. Will be passed to `data.table::setDTthreads`.")
   ),
   inputObjects = bindrows(
-    expectsInput("rasterToMatchLarge", "RasterLayer",
+    expectsInput("rasterToMatchLarge", "SpatRaster",
                  desc = paste("a raster of `studyAreaLarge` in the same resolution and projection the simulation's.",
                               "Defaults to the using the Canadian Forestry Service, National Forest Inventory,",
                               "kNN-derived stand biomass map."),
                  sourceURL = ""),
+    # expectsInput("rawBiomassMap", "SpatRaster",
```

Comment:
why is `rawBiomassMap` removed as  an input? it is used below

Reply:
It's not a "true" input in the sense that it's only (potentiallty) used as a template to RTM during `.inputObjects`.
If it already exists in `sim` then it's used -- to make sure we use the _same_ object if the e.g., the user provides it - if not then a temporary `rawBiomassMap` is produced, but not kept.

BTW ths changed has been in development for over a year now: c36e6110bad1e76184ddae163b7b793dc79557ee